packages/shared/src/components/profile/UserList.tsx

```diff
@@ -14,15 +20,31 @@ export interface UserListProps {
   users: UserShortProfile[];
   placeholderAmount?: number;
   additionalContent?: (user: UserShortProfile, index: number) => ReactNode;
+  squad?: Squad;
 }
 
 function UserList({
   placeholderAmount,
   scrollingProps,
   users,
   additionalContent,
+  squad,
   ...props
 }: UserListProps): ReactElement {
+  const { trackEvent } = useContext(AnalyticsContext);
+  const [copying, copyLink] = useCopyLink(() => {
+    const permalink = squad?.permalink;
+    const token = squad?.currentMember?.referralToken;
+
+    if (!permalink || !token) {
+      return undefined;
+    }
+
+    const invitation = `${permalink}/${token}`;
+
+    return invitation;
+  });
```

Comment:
I think we should maybe refactor this to the top level and simply pass down the function to relevant components.

OR - create a hook for this (useCopySquadLink) since it will be a uniform hook that's used by multiple components.

Let me know what you think.

Reply:
Thought about this. It can be done for sure.

Reply:
I have created a hook in a different PR for copying. It will be able to utilize what you have here if we would want to keep the copying and the invitation separate.